cornflow-server/cornflow/endpoints/login.py

```diff
@@ -137,61 +147,39 @@ def auth_ldap_authenticate(self, username, password):
 
         return user
 
-    def auth_oid_authenticate(
-        self, token: str = None, username: str = None, password: str = None
-    ):
+    def auth_oid_authenticate(self, token: str = None, username: str = None, password: str = None):
         """
-        Method  in charge of performing the log in with the token issued by an Open ID provider.
-        It has an exception and thus accepts username and password for service users if needed.
+        Method in charge of performing the authentication using OpenID Connect tokens.
+        Supports any OIDC provider configured via provider_url.
 
-        :param str token: the token that the user has obtained from the Open ID provider
-        :param str username: the username of the user to log in
-        :param str password: the password of the user to log in
-        :return: the user object or it raises an error if it has not been possible to log in
+        :param str token: the JWT token from the OIDC provider
+        :param str username: username for service users
+        :param str password: password for service users
+        :return: the user object, or it raises an error if it has not been possible to log in
         :rtype: :class:`UserModel`
         """
-
         if token:
-
-            oid_provider = int(current_app.config["OID_PROVIDER"])
-
-            client_id = current_app.config["OID_CLIENT_ID"]
-            tenant_id = current_app.config["OID_TENANT_ID"]
-            issuer = current_app.config["OID_ISSUER"]
-
-            if client_id is None or tenant_id is None or issuer is None:
-                raise ConfigurationError("The OID provider configuration is not valid")
-
-            if oid_provider == OID_AZURE:
-                decoded_token = self.auth_class().validate_oid_token(
-                    token, client_id, tenant_id, issuer, oid_provider
+            decoded_token = self.auth_class().decode_token(token)
+
+            username = decoded_token.get('username')
+            if not username:
+                raise InvalidCredentials(
+                    "Invalid token: missing username claim",
+                    log_txt="Token validation failed: missing username claim",
+                    status_code=400
```

Comment:
This could be problematic as username is not a required claim.

Reply:
Decoded_token function returns a dict "username": "sub". The sub its the required claim